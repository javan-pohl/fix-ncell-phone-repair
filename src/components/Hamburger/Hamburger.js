import * as React from 'react'
import PropTypes from 'prop-types'
import { hamburgerMain, hamburger } from './Hamburger.module.css'

const Hamburger = ({ hamClick }) => (
  <div className={hamburgerMain} onClick={() => hamClick()} onKeyDown={() => hamClick()} role="button" tabIndex={0}>
    <div className={hamburger} />
    <div className={hamburger} />
    <div className={hamburger} />
  </div>
)

Hamburger.propTypes = {
  hamClick: PropTypes.func.isRequired,
}
export default Hamburger
