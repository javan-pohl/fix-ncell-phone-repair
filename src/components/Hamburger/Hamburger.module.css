
.hamburger {
	width: 35px;
  height: 4px;
  background-color: black;
  margin: 6px 30px 0px 0px;
}