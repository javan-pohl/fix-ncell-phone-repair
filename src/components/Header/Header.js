import * as React from 'react'
import { Link } from 'gatsby'
import Navigation from '../Navigation/Navigation'
import { header, title, titleWhite } from './Header.module.css'

function Header() {
  return (
    <header className={header}>
      <div className={title}>
        <Link to="/">
          FixNcell
          {' '}
          <span className={titleWhite}>Phone Repair</span>
        </Link>
      </div>
      <Navigation />
    </header>
  )
}

export default Header
