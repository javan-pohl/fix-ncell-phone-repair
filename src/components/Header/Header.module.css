.header {
  display: flex;
  justify-content: space-between;
	align-items: flex-end;
  width: 100%;
  height: 100%;
  background: #0098ff;
  color: #ffffff;
  padding: 15px 5px;
  min-height: 10px;
  border-bottom: #000000 3px solid;
}
.header a {
  color: black;
  text-decoration: none;
}
.li {
  float: left;
  font-weight: bold;
  color: black;
  padding: 0 0 0 15px;
  margin: 0px;
}
.li a {
  text-transform: uppercase;
}
.li a:hover {
  color: white;
}
.nav {
	margin: 0;
}
.navUl {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.title {
  color: black;
  float: left;
  font-weight: bold;
  font-size: 25px;
}

.titleWhite {
  color: white;
}
