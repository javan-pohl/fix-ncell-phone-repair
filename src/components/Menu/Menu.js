import React, { useImperativeHandle, useRef } from 'react'

import { Link } from 'gatsby'
import { menu, open } from './Menu.module.css'

// import styles from './Menu.module.scss'

class Menu extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      open: false,
    }
  }

  close() {
    this.setState({ open: false })
  }

  open() {
    this.setState({ open: true })
  }

  render() {
    return (
      <div
				id="navMenu"
        role="button"
        tabIndex="0"
/* we don't want to destructure props here because that will cause name overlap */
/* eslint-disable */
        className={`${menu} ${this.state.open ? `${open}` : ''}`}
/* eslint-enable */
        onClick={() => this.close()}
        onKeyDown={() => this.close()}
      >
        <Link to="/contact">
          <h1>Contact Us</h1>
        </Link>
        <Link to="/">
          <h1>Home</h1>
        </Link>
        <Link to="/about">
          <h1>About</h1>
        </Link>
        <Link to="/iphone-repair">
          <h1>Repairs</h1>
        </Link>
      </div>
    )
  }
}

export default React.forwardRef((props, ref) => {
  const menuRef = useRef()

  useImperativeHandle(ref, () => ({
    open() {
      menuRef.current.open()
    },
  }))

  return <Menu ref={menuRef} {...props} />
})
