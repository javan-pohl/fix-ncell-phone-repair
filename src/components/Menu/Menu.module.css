.menu {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #ffffff;
  z-index: 1031;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  transition: all 0.35s;

  cursor: pointer;
}

a {
  pointer-events: all;
}