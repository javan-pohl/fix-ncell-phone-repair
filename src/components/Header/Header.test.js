import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from 'react'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import { shallow, mount, configure } from 'enzyme'

import Header from './Header'
import Hamburger from '../Hamburger/Hamburger'
import Menu from '../Menu/Menu'

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

// configure({ adapter: new Adapter() })

// describe('Test Hamburger Menu', () => {
//   it('Test click event', () => {
//     // const hambMenu = shallow((<Header />))
//     const wrapper = mount(<Header />)
// 		const propsChildren = wrapper.props.children
//     const navChildren = wrapper.children()
// 		const navMenu = wrapper.find('#navMenu')
// 		console.log('navChildren: ', navChildren)
// 		console.log('propsChildren: ', propsChildren)
// 		console.log('wrapper: ', wrapper)
//     // navChildren.find('#hamburgerButton').simulate('click')
//     wrapper.find(Hamburger).simulate('click')
//     expect(wrapper.find(Menu)).toBeVisible()
//   })
// })