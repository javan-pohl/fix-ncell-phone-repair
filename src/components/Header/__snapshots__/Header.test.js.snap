// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="title"
  >
    <a
      href="/"
    >
      FixNcell
       
      <span
        className="titleWhite"
      >
        Phone Repair
      </span>
    </a>
  </div>
  <div
    className="hamburgerMain"
    id="hamburgerButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="hamburger"
    />
    <div
      className="hamburger"
    />
    <div
      className="hamburger"
    />
  </div>
  <div
    className="menu "
    id="navMenu"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <a
      href="/contact"
    >
      <h1>
        Contact Us
      </h1>
    </a>
    <a
      href="/"
    >
      <h1>
        Home
      </h1>
    </a>
    <a
      href="/about"
    >
      <h1>
        About
      </h1>
    </a>
    <a
      href="/iphone-repair"
    >
      <h1>
        Repairs
      </h1>
    </a>
  </div>
</header>
`;
